File: aoc2024/src/test/kotlin/Day14Test.kt
import common.Vec
import day14.*
import kotlin.test.Test
import kotlin.test.assertEquals

class Day14Test {
    @Test
    fun `Input parsing works`() {
        val parsed = parseInput(streamInput("14.txt"))
        val expected = listOf(
            Robot(Vec(0, 4), Vec(3, -3)),
            Robot(Vec(6, 3), Vec(-1, -3)),
            Robot(Vec(10, 3), Vec(-1, 2)),
            Robot(Vec(2, 0), Vec(2, -1)),
            Robot(Vec(0, 0), Vec(1, 3)),
            Robot(Vec(3, 0), Vec(-2, -2)),
            Robot(Vec(7, 6), Vec(-1, -3)),
            Robot(Vec(3, 0), Vec(-1, -2)),
            Robot(Vec(9, 3), Vec(2, 3)),
            Robot(Vec(7, 3), Vec(-1, 2)),
            Robot(Vec(2, 4), Vec(2, -3)),
            Robot(Vec(9, 5), Vec(-3, -3)),
        )
        assertEquals(expected, parsed)
    }

    @Test
    fun `Part 1 example works`() {
        val robots = parseInput(streamInput("14.txt"))
        assertEquals(12, part1(Room(robots, Vec(11, 7))))
    }
}

File: 2022/src/test/scala/Day14Test.scala
import org.scalatest.funsuite.AnyFunSuite
import day14.{Point, Solution}

class Day14Test extends AnyFunSuite {
  private val input =
    """
      |498,4 -> 498,6 -> 496,6
      |503,4 -> 502,4 -> 502,9 -> 494,9
      |""".stripMargin

  test("Solution.parseInput") {
    val expected = Set(
      Point(498, 4), Point(498, 5),Point(498, 6), Point(497, 6), Point(496, 6),
      Point(503, 4), Point(502, 4), Point(502, 5), Point(502, 6), Point(502, 7),
      Point(502, 8), Point(502, 9), Point(501, 9), Point(500, 9), Point(499, 9),
      Point(498, 9), Point(497, 9), Point(496, 9), Point(495, 9), Point(494, 9),
    )
    assert(Solution.parseInput(io.Source.fromString(input)) === expected)
  }
}
